### PR TITLE
Adding minor fix where any failure call to publish telemetry should be ignored

### PR DESCRIPTION
**Pull request by nigurr**

### CODE REVIEW

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs [hunk 1]
```diff
@@ -82,13 +82,20 @@ private void ProcessPublishTelemetryCommand(IExecutionContext context, Dictionar
                 throw new ArgumentException(StringUtil.Loc("TelemetryCommandDataError", data, ex.Message));
             }
 
-            var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
-            var vssConnection = WorkerUtilities.GetVssConnection(context);
-            ciService.Initialize(vssConnection);
+            try
```
Comment (TingluoHuang) [inline src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: i think you might want to make the entire `ProcessPublishTelemetryCommand` best effort?
change any exception throw to be warning within this method

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs [hunk 2]
```diff
@@ -82,13 +82,20 @@ private void ProcessPublishTelemetryCommand(IExecutionContext context, Dictionar
...
+                var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
+                var vssConnection = WorkerUtilities.GetVssConnection(context);
+                ciService.Initialize(vssConnection);
 
-            var commandContext = HostContext.CreateService<IAsyncCommandContext>();
-            commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
-            commandContext.Task = PublishEventsAsync(context, ciService, ciEvent);
+                var commandContext = HostContext.CreateService<IAsyncCommandContext>();
+                commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
+                commandContext.Task = PublishEventsAsync(context, ciService, ciEvent);
```
Comment (TingluoHuang) [inline src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: i don't think these 3 lines of code will throw exception, you can take them out of the try-catch
```
var commandContext = HostContext.CreateService<IAsyncCommandContext>();
commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
commandContext.Task = PublishEventsAsync(context, ciService, ciEvent);
```

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs (hunk 1 above)
Comment (nigurr) [reply src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: I thought about that. but the area and feature other things are valid exception and best to be thrown. only calling http function should be ignore incase of error

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs (hunk 2 above)
Comment (nigurr) [reply src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: I just let it to be safe this time :)

**Comment by nigurr** [comment]
@TingluoHuang can you please approve the same?

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs (hunk 2 above)
Comment (TingluoHuang) [reply src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: i would still prefer move it out. :)

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs (hunk 1 above)
Comment (TingluoHuang) [reply src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: i am fine with this.

**Comment by TingluoHuang** [comment]
please fix L0 test and address the second comment, then merge.

**Pull request merged**